web-registry/src/lib/ecocredit.ts

```diff
@@ -134,21 +197,36 @@ export const getBatchesWithSupply = async (
   );
 };
 
-export const addSupplyDataToBatch = (
+export const addSupplyDataToBatch = async (
```

Comment:
we could probably rename this function since we're not only adding supply but also tx hash now

Reply:
now `addDataToBatch`- generic but short name